### DISCUSSION TITLE
Refreshing the web browser

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [beeware/toga#1695] [Q&A]
I set up a python webserver on http://localhost:8080 which is serving content.

I used the Toga WebView widget and pointed it at localhost.

This works beautifully on Linux (have not tested on other platforms yet).

I cannot seem to get the browser to refresh using either refresh_sublayouts() or refresh(), but the third option shown below did work. 

Code snippet:
```
self.parent.refresh_sublayouts()
self.parent.children[3].parent.refresh()
self.parent.children[3].url = "http://localhost:8080"
```
Perhaps the refresh() options are not implemented on Linux?

**Comment by mhsmith** [comment]
As it says in [the documentation](https://toga.readthedocs.io/en/latest/reference/api/widgets/widget.html#toga.widgets.base.Widget.refresh), "refresh" in these methods refers to Toga's own layout and appearance. We should probably have a separate method to do a refresh of the web page itself, because requiring the user to reassign the url to the same value isn't intuitive.

**Comment by eddyizm** [comment]
Found this while searching for a solution. the page itself does not respond to ctrl+R for a fresh _ i was thinking i may be able to capture that shortcut and map it to toga. 
maybe a `page_reload()` or `webpage_refresh()` method. to distinguish the webview.

**Reply by freakboy3742** [reply]
I don't think there's a separate method for this; `page_reload()` is really not much more than `self.page.url = self.page.url` - that is, explicitly re-setting the URL on the page to what it already is.
